test/extended/util/framework.go

```diff
@@ -280,3 +281,19 @@ func ExtendedTestPath() string {
 func FixturePath(elem ...string) string {
 	return filepath.Join(append([]string{ExtendedTestPath()}, elem...)...)
 }
+
+// CheckURL grabs the output from the specified url and returns it.
+// It will retry up to N times if an error occurs during the request.
+func CheckURL(url string, retries int) (response string, err error) {
+	r, err := http.Get(url)
+	count := 0
+	for count < retries && err != nil {
```

Comment:
> it has the same functional semantics in terms of being side channel injected and subject to non-obvious manipulation and clobbering. help me out @deads2k

I hate closures, but this fight can't be won.  Javascript has taught people terrible things and now they want them everywhere....